webhooks/clusterclass.go

```diff
@@ -134,15 +133,15 @@ func (webhook *ClusterClass) validate(old, in *clusterv1.ClusterClass) error {
 func (webhook *ClusterClass) validateAllRefs(in *clusterv1.ClusterClass) field.ErrorList {
 	var allErrs field.ErrorList
 
-	allErrs = append(allErrs, webhook.validateTemplate(&in.Spec.Infrastructure, in.Namespace, field.NewPath("spec", "infrastructure"))...)
-	allErrs = append(allErrs, webhook.validateTemplate(&in.Spec.ControlPlane.LocalObjectTemplate, in.Namespace, field.NewPath("spec", "controlPlane"))...)
+	allErrs = append(allErrs, check.TemplateIsValid(&in.Spec.Infrastructure, in.Namespace, field.NewPath("spec", "infrastructure"))...)
```

Comment:
Should this be
```suggestion
	allErrs = append(allErrs, check.TemplateRefIsValid(&in.Spec.Infrastructure, in.Namespace, field.NewPath("spec", "infrastructure"))...)
```
Given that we are validating a reference (not the actual template)